Program.fs

```diff
@@ -80,37 +84,25 @@ let needsNodeModules (packageJson: string) =
     not nodeModulesExists
 
 let findInstalledPackages (packageJson: string) : ResizeArray<InstalledNpmPackage> =
-    let content = JObject.Parse(IO.File.ReadAllText packageJson)
-    let dependencies : JProperty list = [
-        if content.ContainsKey "dependencies"
-        then yield! (content.["dependencies"] :?> JObject).Properties() |> List.ofSeq
-    ]
-
-    let devDependencies = [
-        if content.ContainsKey "devDependencies"
-        then yield! (content.["devDependencies"] :?> JObject).Properties() |> List.ofSeq
-    ]
-
-    let topLevelPackages = ResizeArray [
-        yield! [
-            for package in dependencies -> {
-                Name = package.Name;
-                Range = Some (SemVer.Range(package.Value.ToObject<string>()));
-                Installed = None
-                DevDependency = false
-            }
-        ]
-
-        yield! [
-            for package in devDependencies -> {
-                Name = package.Name;
-                Range = Some (SemVer.Range(package.Value.ToObject<string>()));
-                Installed = None
-                DevDependency = true
-            }
-        ]
-
-    ]
+    let file = IO.File.ReadAllText packageJson
+    let topLevelPackages =
+        match Decode.Auto.fromString<RawPackage>(file, isCamelCase = true) with
+        | Ok rawPackage ->
+            let createInstalledPackages isDevDependency (dependencies: Map<string, string> option) =
+                 dependencies
+                 |> Option.defaultValue Map.empty
+                 |> Seq.map (fun pair -> {
+                    Name = pair.Key;
+                    Range = Some (SemVer.Range(pair.Value));
+                    Installed = None;
+                    DevDependency = isDevDependency
+                })
+            
+            ResizeArray [
+                yield! createInstalledPackages false rawPackage.Dependencies;
+                yield! createInstalledPackages true rawPackage.DevDependencies
+            ]
+        | Error _ -> ResizeArray []
```

Comment:
Suggestion: log the parsing error to the console before returning the empty `ResizeArray [ ]`